sources/lib/datalab/gcp/datalab/__init__.py

```diff
@@ -86,3 +94,24 @@ def _run_cell_magic(self, magic_name, line, cell):
 _shell.InteractiveShell.run_cell_magic = _run_cell_magic
 _shell.InteractiveShell.run_line_magic = _run_line_magic
 
+# Define global 'project_id' and 'set_project_id' functions to manage the default project ID. We
+# do this conditionally in a try/catch # to avoid the call to Context.default() when running tests
+# which mock IPython.get_ipython().
+
+def _get_project_id():
+  try:
+    return _context.Context.default().project_id
+  except Exception:
+    return None
+
+def _set_project_id(project_id):
+  _context.Context.default(project_id)
+
+try:
+  if 'project_id' not in _IPython.get_ipython().user_ns:
+    _IPython.get_ipython().user_ns['project_id'] = _get_project_id
```

Comment:
Possibly in a separate change, expose it as a param in datalab.sh so it can be used as default? It's tedious to set it for each notebook.